server/services/omnichannel-voip/service.ts

```diff
@@ -72,4 +72,41 @@ export class OmnichannelVoipService extends ServiceClass implements IOmnichannel
 			})),
 		};
 	}
+
+	private getQueuesForExt(
+		ext: string,
+		queueInfo: {
+			name: string;
+			members: string[];
+		}[],
+	): string[] {
+		return queueInfo.reduce((acc: string[], queue: { name: string; members: string[] }) => {
+			if (queue.members.includes(ext)) {
+				acc.push(queue.name);
+			}
+			return acc;
+		}, []);
+	}
+
+	async getExtensionListWithAgentData(): Promise<IVoipExtensionWithAgentInfo[]> {
+		const { result: extensions } = await Voip.getExtensionList();
```

Comment:
I am not sure if pagination is possible here. Asterisk itself does not provide any pagination. https://wiki.asterisk.org/wiki/display/AST/Asterisk+16+ManagerAction_PJSIPShowEndpoints
What we will achieve by in-memory pagination? Furthermore, we do not maintain the extensions. So if someone on asterisk's side changes the extension map, the pagination logic would go wrong. e.g